test/services/ContractorDetailsServiceSpec.scala

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import models.ContractorDetailsViewModel
+import models.mongo.{CisCYAModel, CisUserData, DatabaseError}
+import org.joda.time.{DateTime, DateTimeZone}
+import support.builders.models.UserBuilder._
+import support.builders.models.mongo.CisCYAModelBuilder.aCisCYAModel
+import support.builders.models.mongo.CisUserDataBuilder._
+import support.mocks.MockCISUserDataRepository
+import support.{TaxYearHelper, UnitTest}
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.TestingClock
+
+import scala.concurrent.ExecutionContext
+
+class ContractorDetailsServiceSpec extends UnitTest
+  with MockCISUserDataRepository
+  with TaxYearHelper {
+
+  implicit private val hc: HeaderCarrier = HeaderCarrier()
+  val clock = TestingClock
+  private val underTest = new ContractorDetailsService(mockRepo, clock)
+  val dateTime = clock.now()
+  implicit val ec = ExecutionContext.global
+
+  val contractorDetailsViewModel = new ContractorDetailsViewModel("ABC Steelworks", "123/AB123456")
+  val newCisUserData = CisUserData(
+    aUser.sessionId,
+    aUser.mtditid,
+    aUser.nino,
+    taxYear,
+    contractorDetailsViewModel.employerReferenceNumber,
+    None,
+    false,
+    (CisCYAModel(Some(contractorDetailsViewModel.contractorName), None)),
+    DateTime.parse("2022-05-11T16:38:57.489Z")
+  )
+
+  ".checkAccessContractorDetailsPage" should {
+    "return left with an error message " in {
+      val error = new DatabaseError {
+        override val message: String = "Database broke"
+      }
+      mockFindCYAData(taxYear, aCisUserData.employerRef,  aUser, Left(error))
+      await(underTest.checkAccessContractorDetailsPage(taxYear, aUser, aCisUserData.employerRef)) shouldBe Left("Database broke")
```

Comment:
you can use one of the database errors instead of a string e.g. DataNotUpdatedError